openstack/openstack_vm.yaml

```diff
@@ -99,29 +99,35 @@ node_types:
-              ssh_config:               { type: string,  default: { get_property: [ SELF, ssh_config ] } }
-              include_exporter:         { type: boolean, default: { get_property: [ SELF, include_exporter ] } }
-              timeout:                  { type: integer, default: { get_property: [ SELF, timeout ] } } 
-              floating_ip_pools:        { type: list,    default: { get_property: [ SELF, floating_ip_pools ] } }
-              userdata:                 { type: string,  default: { get_property: [ SELF, userdata ] } } 
-              env:                      { type: map,     default: { get_property: [ SELF, env ] } } 
-              consul_server_address:    { type: string,  default: { get_property: [ SELF, consul_server_address ] } }
+              network:                  { type: string,  default: { get_property:  [ SELF, network               ] } } 
+              flavor:                   { type: string,  default: { get_property:  [ SELF, flavor                ] } } 
+              key_name:                 { type: string,  default: { get_property:  [ SELF, key_name              ] } } 
+              image:                    { type: string,  default: { get_property:  [ SELF, image                 ] } } 
+              security_groups:          { type: string,  default: { get_property:  [ SELF, security_groups       ] } } 
+              vm_name:                  { type: string,  default: { get_property:  [ SELF, name                  ] } } 
+              ansible_user:             { type: string,  default: { get_property:  [ SELF, username              ] } } 
+              ssh_config:               { type: string,  default: { get_property:  [ SELF, ssh_config            ] } }
+              include_exporter:         { type: boolean, default: { get_property:  [ SELF, include_exporter      ] } }
+              timeout:                  { type: integer, default: { get_property:  [ SELF, timeout               ] } } 
+              floating_ip_pools:        { type: list,    default: { get_property:  [ SELF, floating_ip_pools     ] } }
+              userdata:                 { type: string,  default: { get_property:  [ SELF, userdata              ] } } 
+              env:                      { type: map,     default: { get_property:  [ SELF, env                   ] } } 
+              consul_server_address:    { type: string,  default: { get_property:  [ SELF, consul_server_address ] } }
+              deployment_label:         { type: string,  default: { get_input:             deployment_label        } }
+              monitoring_id:            { type: string,  default: { get_input:             monitoring_id           } }
+              instance_name:            { type: string,  default: { get_attribute: [ SELF, tosca_name            ] } }
+              id:                       { type: string,  default: { get_attribute: [ SELF, id                    ] } } 
```

Comment:
`id` is an attribute, obtained inside [vm_create.yml](https://github.com/SODALITE-EU/iac-modules/blob/4ab3e7aa07728810179f62a55a8d7b9a1b88ad7b/openstack/playbooks/vm_create.yml#L46) playbook. You don't need to pass it as an input, neither you should. It produces an error ("Cannot use an unset value: id")

Reply:
This line needs to be removed